exts/cesium.powertools/cesium/powertools/ui/georef_helper_window.py

```diff
@@ -0,0 +1,149 @@
+import logging
+import omni.ui as ui
+import omni.usd
+from ..utils.proj import epsg_to_ecef, epsg_to_wgs84, get_crs_name_from_epsg
+import math
+from .custom_fields import string_field_with_label, int_field_with_label, float_field_with_label
+
+
+class CesiumGeorefHelperWindow(ui.Window):
+    WINDOW_NAME = "Cesium Georeference Helper"
+
+    _logger: logging.Logger
+
+    def __init__(self, **kwargs):
+        super().__init__(CesiumGeorefHelperWindow.WINDOW_NAME, **kwargs)
+
+        self._logger = logging.getLogger(__name__)
+
+        # Set the function that is called to build widgets when the window is visible
+        self.frame.set_build_fn(self._build_fn)
+
+    def destroy(self):
+        # It will destroy all the children
+        super().destroy()
+
+    def __del__(self):
+        self.destroy()
+
+    def convert_coordinates(self):
+        # Get the CRS and check if it is valid, adjust UI values accordingly
+        crs = get_crs_name_from_epsg(self._epsg_model.get_value_as_int())
+        if crs is None:
+            self._epsg_name_model.set_value("Invalid EPSG Code")
+            self._wgs84_latitude_model.set_value(math.nan)
+            self._wgs84_longitude_model.set_value(math.nan)
+            self._wgs84_height_model.set_value(math.nan)
+            self._ecef_x_model.set_value(math.nan)
+            self._ecef_y_model.set_value(math.nan)
+            self._ecef_z_model.set_value(math.nan)
```

Comment:
I think this is maybe the second or third time I've actually seen someone use `math.nan` correctly, so bravo.